.github/workflows/selenium_tests.yml

```
@@ -0,0 +1,28 @@
+name: Selenium Tests
+
+on:
+  pull_request:
+    types: [ labeled, synchronize, opened, reopened ]
+
+permissions:
+   contents: read
+
+jobs:
+  Run_Selenium_tests:
+    if: contains(github.event.pull_request.labels.*.name, 'enhancement')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+          cache: 'pip'
+          cache-dependency-path: 'tests/requirements/selenium.txt'
+      - name: Install and upgrade packaging tools
+        run: python -m pip install --upgrade pip setuptools wheel
+      - run: python -m pip install -r tests/requirements/py3.txt .
+
+      - name: Run selenium tests for current PR
```

"for current PR" seems redundant:
```suggestion
      - name: Run selenium tests
```